flow/scenarios/highway/gen.py

```diff
@@ -20,30 +21,34 @@ def __init__(self, net_params, base):
     def specify_nodes(self, net_params):
         """See parent class."""
         length = net_params.additional_params["length"]
+        num_edges = net_params.additional_params.get("num_edges", 1)
+        segment_lengths = np.linspace(0, length, num_edges+1)
 
-        nodes = [{
-            "id": "begin",
-            "x": repr(0),
-            "y": repr(0)
-        }, {
-            "id": "end",
-            "x": repr(length),
-            "y": repr(0)
-        }]
+        nodes = []
+        for i in range(num_edges+1):
+            nodes += [{
+                "id": "edge_{}".format(i),
+                "x": repr(segment_lengths[i]),
+                "y": repr(0)
+            }]
 
         return nodes
 
     def specify_edges(self, net_params):
         """See parent class."""
         length = net_params.additional_params["length"]
-
-        edges = [{
-            "id": "highway",
-            "type": "highwayType",
-            "from": "begin",
-            "to": "end",
-            "length": repr(length)
-        }]
+        num_edges = net_params.additional_params.get("num_edges", 1)
+        segment_length = length/float(num_edges)
+
+        edges = []
+        for i in range(num_edges):
```

Comment:
sort of at a high level, why do we want to cut the highway into sections?

Reply:
This lets me write tests for the methods that get leaders for subsequent edges